homework/unique_ptr/UniquePtr.hpp

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include <iostream>
+
+struct NullPtrDereferenceException : public std::runtime_error {
+    NullPtrDereferenceException()
+        : std::runtime_error{"Fatal error, cannot dereference nullptr"} {}
+};
+
+template <typename T>
+class UniquePtr {
+public:
+    UniquePtr(T* ptr);
+
+    UniquePtr(const UniquePtr<T>&) = delete;
+    UniquePtr<T>& operator=(const UniquePtr<T>&) = delete;
+
+    UniquePtr(UniquePtr<T>&& ptr);
+    UniquePtr<T>& operator=(UniquePtr<T>&& ptr);
+
+    T operator*() const;
```

Comment:
```suggestion
    T& operator*() const;
```
This operator should return a reference, not a copy